### ISSUE TITLE
Prepare for DAG based workflows

### ISSUE THREAD

**Issue by lalitpagaria** [lalitpagaria/obsei#145]
Working towards full DAG base workflows https://github.com/lalitpagaria/obsei/issues/8

Following things are part of these Epic -
- [ ] Introduce generic `node` structure
- [ ] Introduce `node_type` structure (Observer, Informer, Analyzer, 
- [ ] Replace use of `Sink` with `Informer` (packages, classes and variables)
- [ ] Replace use of `Source` with `Observer` (packages, classes and variables)
- [ ] Replace use of `Analyzer` with `Segmenter` (packages, classes and variables)

**Comment by lalitpagaria** [comment]
Let's plan first version for release v0.1. This release will have many breaking changes hence better to increase minor version.

If we can leverage UI of https://github.com/orchest/orchest then it will be nice addition to Obsei.

**Comment by shahrukhx01** [comment]
@lalitpagaria https://github.com/orchest/orchest are using typescript, we need to make a decision about how we want to proceed with the UI. We may have to do some changes to adapt Orchest, that may avert our focus on Obsei itself. Second option is either continue with streamlit or use Justpy. Drag and drop interfaces are possible with Justpy and tailwind, for instance this example can easily be replicated with Justpy https://github.com/ananikets18/Drag-and-Drop-card-using-HTML-TailwindCSS-

**Comment by lalitpagaria** [comment]
@shahrukhx01 I agree that UI is not top priority for platform point of view. But I feel it will be good for user adoption point of view (I my view, but I might be wrong). 

In past I experiment with [streamlit](https://share.streamlit.io/lalitpagaria/obsei/sample-ui/ui.py) but it did not clicked. Yes one point is to host demo on our own server (Now it is possible using Oracle cloud), with proper rate limiting to prevent users from abusing demo.

I also tried to build demo using Justpy but I felt it too complex. It could be due to that fact that front end is not my strong area. If you have experience, is it possible to prepare drag and drop based workflow creator? Main idea to generate workflow yaml for user. For first version we can support 1 time execution.

**Comment by shahrukhx01** [comment]
@lalitpagaria I can try creating a simple UI demo using Justpy once I get time. Meanwhile, we can decide how we want to design our backend codebase since that's more important.

**Comment by lalitpagaria** [comment]
Above issue for backend codebase only. For UI let's create separate issue :)

**Comment by shahrukhx01** [comment]
@lalitpagaria sure you can assign me that.

**Comment by GirishPatel** [comment]
For creating architecture diagram we can use [Archimate](https://www.visual-paradigm.com/tutorials/how-to-draw-archimate-diagram.jsp).
We can use [PlantUML](https://plantuml.com/archimate-diagram) for create diagram.
